usecases/replica/coordinator.go

```diff
@@ -35,33 +41,43 @@ type (
 	// coordinator coordinates replication of write and read requests
 	coordinator[T any] struct {
 		Client
-		Resolver *resolver // node_name -> host_address
-		log      logrus.FieldLogger
-		Class    string
-		Shard    string
-		TxID     string // transaction ID
+		Resolver    *resolver // node_name -> host_address
+		log         logrus.FieldLogger
+		Class       string
+		Shard       string
+		TxID        string // transaction ID
+		pullBackOff pullBackOffConfig
+	}
+	// pullBackOffConfig controls the op/worker backoffs in coordinator.Pull
+	pullBackOffConfig struct {
```

Comment:
**suggestion:** we could use https://github.com/cenkalti/backoff instead. it's easy to configure for constant and exponential backoffs and we can decrease the logic for `retry`, `backoff` etc.

Reply:
Thanks, I just swapped out the old hand written backoff function, but couldn't find a way to use the `backoff.Retry` function since there are potentially more hosts than workers and I want to keep track of retries on a per-host basis. Let me know if you can think of a way to simplify more